### ISSUE TITLE
React Native Formik: Maximum call stack size exceeded on `onChange` function

### ISSUE THREAD

**Issue by vilvaathibanpb** [jaredpalmer/formik#3823]
### Current Behavior

When i used the latest formik (2.4.1) with the latest RN (0.71.8), I am facing a strange issue. Whenever `handleChange` or `handleSubmit` is executed, there is an infinite loop. 

With debugging I found issue in this [line](https://github.com/jaredpalmer/formik/blob/main/packages/formik/src/Formik.tsx#L582). 
When validation is ignored, by checking `shouldValidate === true`, the issue is gone. 

I also tried debugging validations, but `fieldRegistry` ref stays empty and hence no validations are returned in the `validateFormWithHighPriority` function. 

### Expected behavior

It should work without entering infinite loop

### Your environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Formik           | 2.4.1
| React            | 18.2.0
| TypeScript       |
| Browser          | 
| npm/Yarn         | yarn
| Operating System | Android (RN)

**Comment by aandronache** [comment]
This issue makes the latest version completely unusable 😞

**Comment by aandronache** [comment]
This issue makes the latest versions completely unusable 😞

**Comment by fum4** [comment]
This issue makes the latest version completely unusable 😞

**Comment by Mihai1iMob** [comment]
> This issue makes the latest version completely unusable 😞

Indeed, because of this, I had to revert to `formik@2.3.3` to be able to use it again. :(
